libraries/extractor/src/main/java/androidx/media3/extractor/mp4/AtomParsers.java

```diff
@@ -1237,6 +1247,22 @@ private static void parseVideoSampleEntry(
       } else if (childAtomType == Atom.TYPE_av1C) {
         ExtractorUtil.checkContainerInput(mimeType == null, /* message= */ null);
         mimeType = MimeTypes.VIDEO_AV1;
+        parent.setPosition(childStartPosition + Atom.HEADER_SIZE);
+        if (childAtomSize > Atom.HEADER_SIZE) {
```

Comment:
What is the reason behind using this as the condition to see if an AV1CodecConfigurationBox is included?

Reply:
This comes from a fix we made in our fork, where we found encodes that had missing HEVCDecoderConfigurationRecord so we added that check for all such atoms. However given that the specs, both ISO 14496-15 and the AV1 one, say "sample entry **shall** contain an ...Configuration Box" it seems such encodes would not be compliant.
Long story short I think you're right, I should remove the check at line 1251.